src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesFlowsV2Spec.groovy

```diff
@@ -63,14 +67,14 @@ class SwitchesFlowsV2Spec extends HealthCheckSpecification {
         switchPair = switchPairs.all()
                 .includeSwitch(switchTriplet.getShared())
                 .includeSwitch(switchTriplet.getEp1()).random()
-        def flowDefinition = flowHelperV2.randomFlow(switchPair, false).tap { allocateProtectedPath = true }
-        flowId = flowHelperV2.addFlow(flowDefinition, UP, CLASS).getFlowId()
-        switchFlowGoesThrough = pathHelper.getInvolvedSwitches(flowId).find {
-            ![switchPair.getSrc(), switchPair.getDst()].contains(it)
-        }
-        switchProtectedPathGoesThrough = pathHelper.getInvolvedSwitchesForProtectedPath(flowId).find {
-            ![switchPair.getSrc(), switchPair.getDst()].contains(it)
-        }
+        flow = flowFactory.getBuilder(switchPair, false)
+                .withProtectedPath(true)
+                .build().create(UP, CLASS)
+        flowId = flow.flowId
+        def switchIdFlowGoesThrough = flow.retrieveAllEntityPaths().flowPath.path.forward.getInvolvedSwitches().unique()[1]
```

Comment:
It seems like the logic of switches selection has been changed(transit switch should be selected).
```
def flowPathInfo = flow.retrieveAllEntityPaths()
switchFlowGoesThrough =  topology.activeSwitches.find { it.dpId == flowPathInfo.flowPath.path.forward.transitInvolvedSwitches.first() }
switchProtectedPathGoesThrough =  topology.activeSwitches.find { it.dpId == flowPathInfo.flowPath.protectedPath.forward.transitInvolvedSwitches.first() }
```

Reply:
Done